From 2a1bff244d89c6ff15cbe7553564eb45f90997ac Mon Sep 17 00:00:00 2001
From: Nikolaus Schulz <schulz@macnetix.de>
Date: Thu, 23 Aug 2012 21:46:14 +0200
Subject: [PATCH] Fix debdiff exit code (regression)

Commit 9cbe605 "debdiff: fix CVE-2012-2012 [...]" broke the exit code of
debdiff, at it made wdiff_control_files return 1 if wdiff found no
differences in the control files, which in turn made debdiff exit with an
exit code of 1.
---
 scripts/debdiff.pl |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/scripts/debdiff.pl b/scripts/debdiff.pl
index 7857836..20a0ced 100755
--- a/scripts/debdiff.pl
+++ b/scripts/debdiff.pl
@@ -1034,6 +1034,7 @@ sub wdiff_control_files($$$$$)
 		my $msg = ucfirst($cf) . " files$usepkgname: wdiff output";
 		print "\n", $msg, "\n", '-' x length $msg, "\n";
 		print $wdiff;
+		$status = 1;
 	    } else {
 		my @output;
 		@output = split /\n/, $wdiff;
@@ -1041,8 +1042,8 @@ sub wdiff_control_files($$$$$)
 		my $msg = ucfirst($cf) . " files$usepkgname: lines which differ (wdiff format)";
 		print "\n", $msg, "\n", '-' x length $msg, "\n";
 		print join("\n",@output), "\n";
+		$status = 1;
 	    }
-	    $status = 1;
 	}
     }
 
-- 
1.7.2.5

